src/Microsoft.VisualStudio.Composition/ByValueEquality+AssemblyNameComparer.cs

```diff
@@ -51,29 +51,26 @@ public bool Equals(AssemblyName x, AssemblyName y)
                 // compare the PublicKeys first, but then fall back to GetPublicKeyToken(), which
                 // will generate a public key token for the AssemblyName that has a public key and
                 // return the public key token for the other AssemblyName.
-                byte[] xPublicKey = x.GetPublicKey();
-                byte[] yPublicKey = y.GetPublicKey();
+                byte[]? xPublicKey = x.GetPublicKey();
+                byte[]? yPublicKey = y.GetPublicKey();
 
                 // Testing on FullName is horrifically slow.
                 // So test directly on its components instead.
                 if (xPublicKey != null && yPublicKey != null)
                 {
                     return x.Name == y.Name
-                        && x.Version.Equals(y.Version)
+                        && Equals(x.Version, y.Version)
                         && string.Equals(x.CultureName, y.CultureName)
                         && ByValueEquality.Buffer.Equals(xPublicKey, yPublicKey);
                 }
 
                 return x.Name == y.Name
-                    && x.Version.Equals(y.Version)
+                    && Equals(x.Version, y.Version)
                     && string.Equals(x.CultureName, y.CultureName)
                     && ByValueEquality.Buffer.Equals(x.GetPublicKeyToken(), y.GetPublicKeyToken());
             }
 
-            public int GetHashCode(AssemblyName obj)
-            {
-                return obj.Name.GetHashCode();
-            }
+            public int GetHashCode(AssemblyName obj) => obj.Name?.GetHashCode() ?? 0;
```

Comment:
Is AssemblyName.Name ever null? I wasn't sure from the docs

Reply:
I've never seen it `null` nor expect it to ever be. But [the annotations](https://source.dot.net/#System.Private.CoreLib/AssemblyName.cs,917bceeed23fe2ae) on `netcoreapp3.1` say it could be, so we have to check or use `!` to suppress the warning.